### PR TITLE
Add docs for Spark 3.0.0-preview

### PR DESCRIPTION
**Pull request by jiangxb1987**
This PR adds the voted Spark 3.0.0-preview docs to the website, as the release process required. 

I'll make another PR to update the download page, news post, and documentation.md .

### CODE REVIEW

**Comment by jiangxb1987** [comment]
cc @srowen @dongjoon-hyun @HyukjinKwon @gatorsmile @cloud-fan

File: documentation.md [hunk 1]
```diff
@@ -58,6 +58,7 @@ navigation:
 
 <ul>
   <li><a href="{{site.baseurl}}/docs/2.0.0-preview/">Spark 2.0.0 preview</a></li>
+  <li><a href="{{site.baseurl}}/docs/3.0.0-preview/">Spark 3.0.0 preview</a></li>
```
Comment (srowen) [inline documentation.md]: You'll need to restore this by uncommenting it. I guess it's OK to bring back the 2.0.0 preview link as we have every other version here. See comment below about distinguishing preview releases though. Maybe we can just put them above.

File: js/downloads.js [hunk 2]
```diff
@@ -14,14 +14,18 @@ function addRelease(version, releaseDate, packages, mirrored) {
...
 var scala2p12_hadoopFree = {pretty: "Pre-built with Scala 2.12 and user-provided Apache Hadoop", tag: "without-hadoop-scala-2.12"};
 
 // 2.2.0+
 var packagesV8 = [hadoop2p7, hadoop2p6, hadoopFree, sources];
 // 2.4.0+
 var packagesV9 = [hadoop2p7, hadoop2p6, hadoopFree, scala2p12_hadoopFree, sources];
+// 3.0.0+
+var packagesV10 = [hadoop2p7, hadoop3p2, hadoopFree, sources];
 
+addRelease("3.0.0-preview", new Date("11/06/2019"), packagesV10, true);
```
Comment (srowen) [inline js/downloads.js]: BTW, we used to have a mechanism on this page for showing the 'preview' releases separately. I removed it in https://github.com/apache/spark-website/commit/31485a30bc25e7333a8e698896b5bc05d22f994f as I don't know if it's that important to distinguish.

Other reviewers, am I forgetting a reason we really wanted to separate them?

**Review by srowen — CHANGES_REQUESTED** [review]

**Comment by jiangxb1987** [comment]
Thanks @srowen ! Maybe I shall revert the changes made to the download page and documentation.md to make this PR move fast?

**Comment by srowen** [comment]
Eh I don't think you have to do anything but restore the link to the preview docs, at the minimum. The rest could be another time, I guess.

**Comment by jiangxb1987** [comment]
Errr sorry @srowen I already reverted that. Should it be fine because I'll submit another PR anyway ? After all they are all simple and straightforward changes.

**Comment by srowen** [comment]
OK, if this is just adding the raw docs files, we can link to them next. It's pretty hard to review but skimming a few files looks roughly right.

**Comment by cloud-fan** [comment]
LGTM

**Comment by jiangxb1987** [comment]
Thanks @srowen @cloud-fan !

**Pull request merged**